`examples/with-stripe-typescript` `npm install` fails with 'Could not resolve dependency: npm ERR! peer react@"^16.11.0" from swr@0.1.18'

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```
$ node --version ; npm --version
v16.18.0
8.19.2
```

### What browser are you using? (if relevant)

_No response_

### How are you deploying your application? (if relevant)

_No response_

### Describe the Bug

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: undefined@undefined
npm ERR! Found: react@17.0.2
npm ERR! node_modules/react
npm ERR!   react@"^17.0.2" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.11.0" from swr@0.1.18
npm ERR! node_modules/swr
npm ERR!   swr@"^0.1.16" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See .npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     .npm/_logs/2022-10-19T04_13_23_398Z-debug-0.log
```

### Expected Behavior

Install without error

### Link to reproduction

N/A

### To Reproduce

```sh
$ git clone --depth=1 https://github.com/vercel/next.js -b canary
$ cd next.js/examples/with-stripe-typescript
$ npm i
```

--- a/examples/with-stripe-typescript/components/Cart.tsx
+++ b/examples/with-stripe-typescript/components/Cart.tsx
@@ -1,5 +1,5 @@
 import React, { ReactNode } from 'react'
-import { CartProvider } from 'use-shopping-cart/react'
+import { CartProvider } from 'use-shopping-cart'
 import * as config from '../config'
 
 const Cart = ({ children }: { children: ReactNode }) => (

--- a/examples/with-stripe-typescript/components/CartSummary.tsx
+++ b/examples/with-stripe-typescript/components/CartSummary.tsx
@@ -2,7 +2,7 @@ import React, { useState, useEffect } from 'react'
 
 import StripeTestCards from '../components/StripeTestCards'
 
-import { useShoppingCart } from 'use-shopping-cart/react'
+import { useShoppingCart } from 'use-shopping-cart'
 import { fetchPostJSON } from '../utils/api-helpers'
 
 const CartSummary = () => {

--- a/examples/with-stripe-typescript/components/ClearCart.tsx
+++ b/examples/with-stripe-typescript/components/ClearCart.tsx
@@ -1,5 +1,5 @@
 import { useEffect } from 'react'
-import { useShoppingCart } from 'use-shopping-cart/react'
+import { useShoppingCart } from 'use-shopping-cart'
 
 export default function ClearCart() {
   const { clearCart } = useShoppingCart()

--- a/examples/with-stripe-typescript/components/Products.tsx
+++ b/examples/with-stripe-typescript/components/Products.tsx
@@ -1,6 +1,6 @@
 import products from '../data/products'
 import { formatCurrencyString } from 'use-shopping-cart'
-import { useShoppingCart } from 'use-shopping-cart/react'
+import { useShoppingCart } from 'use-shopping-cart'
 
 const Products = () => {
   const { addItem, removeItem } = useShoppingCart()

--- a/examples/with-stripe-typescript/package.json
+++ b/examples/with-stripe-typescript/package.json
@@ -6,22 +6,23 @@
     "start": "next start"
   },
   "dependencies": {
-    "@stripe/react-stripe-js": "1.7.0",
-    "@stripe/stripe-js": "1.22.0",
-    "micro": "^9.3.4",
+    "@stripe/react-stripe-js": "1.13.0",
+    "@stripe/stripe-js": "1.42.0",
+    "micro": "^9.4.1",
     "micro-cors": "^0.1.1",
     "next": "latest",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2",
-    "stripe": "8.200.0",
-    "swr": "^0.1.16",
-    "use-shopping-cart": "3.0.5"
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
+    "stripe": "10.14.0",
+    "swr": "^1.3.0",
+    "use-shopping-cart": "3.1.2",
+    "redux": "4.2.0"
   },
   "devDependencies": {
-    "@types/micro": "^7.3.3",
-    "@types/micro-cors": "^0.1.0",
-    "@types/node": "^13.1.2",
-    "@types/react": "^16.9.17",
-    "typescript": "4.5.5"
+    "@types/micro": "^7.3.7",
+    "@types/micro-cors": "^0.1.2",
+    "@types/node": "^18.11.2",
+    "@types/react": "^18.0.21",
+    "typescript": "4.8.4"
   }
 }

